src/utils/response/responseHandlers.ts

```diff
@@ -47,41 +64,79 @@ export async function toApiResponse<T>(promise: RequestPromise<T>): Promise<ApiR
 }
 
 /**
- * @return A promise that resolves to T or rejects if no results
+ * @return A promise that resolves to Type or rejects if no results
  */
-export async function required<T>(promise: RequestPromise<T[]>): Promise<T> {
-  const response = await toApiResponse<T[]>(promise);
+export async function required<ResourceName extends ResourceNames>(
+  promise: RequestPromise<ResourceName>
+): Required<ResourceName, Resources[ResourceName][]> {
```

Comment:
Maybe using the Entity type we have there everywhere instead of `Resources[ResourceName]` is more readable 

```suggestion
): Required<ResourceName, Entity<ResourceName>[]> {
```